lib/bypass/instance.ex

```diff
@@ -81,7 +81,7 @@ defmodule Bypass.Instance do
   end
   defp do_handle_call(:down, from, %{socket: socket, ref: ref} = state) when not is_nil(socket) do
     nil = state.caller_awaiting_down  # assertion
-    if state.retained_plug != nil do
+    if state.retained_plugs != [] do
```

Comment:
Calling `down()` on multiple Bypass instances concurrently will trigger the assertion above. `caller_awaiting_down` should also be turned into a list and handled property in `dispatch_awaiting_caller()`.